runtime/common/src/precompile/input.rs

```diff
@@ -186,6 +187,21 @@ where
 
 		Ok(bytes.to_vec())
 	}
+
+	fn bool_at(&self, index: usize) -> Result<bool, Self::Error> {
+		let param = self.u64_at(index)?;
+		if param.is_one() {
+			Ok(true)
+		} else if param.is_zero() {
+			Ok(false)
+		} else {
+			Err(PrecompileFailure::Revert {
+				exit_status: ExitRevert::Reverted,
+				output: "failed to decode bool".into(),
+				cost: self.target_gas.unwrap_or_default(),
+			})
+		}
```

Comment:
I think `param.try_into().map_err(|_| PrecompileFailure::Revert {` is enough.

Reply:
I agree but unfortunately TryFrom<bool> is not implemented for unsigned integers. I also feel like it should be simpler but I don't know of a better way than if statement